src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -694,7 +730,7 @@ class DetailsCard extends React.Component {
                 <Col className={style['fields-column']}>
                   <Grid>
                     <FieldRow label={msg.host()} id={`${idPrefix}-host`}>
-                      { hostName || <NotAvailable tooltip={msg.notAvailableUntilRunning()} id={`${idPrefix}-host-not-available`} /> }
+                      { <span className={style['hostname']} title={this.hostOverflow ? hostName : ''} ref={this.hostRef}>{hostName}</span> || <NotAvailable tooltip={msg.notAvailableUntilRunning()} id={`${idPrefix}-host-not-available`} /> }
```

Comment:
If you extracted a component, this could look like:

````jsx
  <TooltipSpan className={style['hostname']} tooltip={hostName}>
    {hostname}
  </TooltipSpan>
````

And put all the ref, and overflow checking in `TooltipSpan` (or something similarly named)